### ISSUE TITLE
Solitare example script hits exception for "Sprite is already in the list" 

### ISSUE THREAD

**Issue by mriswithe** [pythonarcade/arcade#951]
## Bug Report
Copy/pasted https://api.arcade.academy/en/latest/tutorials/card_game/solitaire_11.html#solitaire-11 into a file (`opus_solitaire.py`) 
When run, I get an exception. Works with Arcade 2.5.7, but 2.6.0 and 2.6.1 both break on the same error shown below in `actual behavior`. Actual index position is different every run since it is part of the shuffling that is going wrong here.

### Tested with:
OS: Windows 10
Arcade:
* 2.6.1
* 2.6.0
* 2.5.7 (This works)

Python:
* 3.9.7
* 3.8.6

### Actual behavior:
```
C:\Users\veigar\PycharmProjects\opus_solitare\venv_38\Scripts\python.exe C:/Users/veigar/PycharmProjects/opus_solitare/opus_solitaire.py
Traceback (most recent call last):
  File "C:/Users/veigar/PycharmProjects/opus_solitare/opus_solitaire.py", line 412, in <module>
    main()
  File "C:/Users/veigar/PycharmProjects/opus_solitare/opus_solitaire.py", line 407, in main
    window.setup()
  File "C:/Users/veigar/PycharmProjects/opus_solitare/opus_solitaire.py", line 181, in setup
    self.card_list[pos1], self.card_list[pos2] = self.card_list[pos2], \
  File "C:\Users\veigar\PycharmProjects\opus_solitare\venv_38\lib\site-packages\arcade\sprite_list\sprite_list.py", line 208, in __setitem__
    raise Exception(f"Sprite is already in the list (index {existing_index})")
Exception: Sprite is already in the list (index 23)

Process finished with exit code 1
```
### Expected behavior:
Solitaire game runs

### Steps to reproduce/example code:
Windows 10 (May not be necessary but didn't check it out on Linux/Mac)
Python 3.9.7
pip install arcade==2.6.1 # Or 2.6.0
Copy the Solitaire example to your machine
Execute it
Receive stacktrace

**Comment by pvcraven** [comment]
Currently we can't swap sprites with:

`self.card_list[pos1], self.card_list[pos2] = self.card_list[pos2], self.card_list[pos1]`

**Comment by einarf** [comment]
The new spritelist implementation can definitely need a couple of improvements. It was a sacrifice to boost performance.

**Comment by mriswithe** [comment]
Possibly dumb questions as I have very little knowledge of this library at this point:

would a `SpriteList.swap(idx1, idx2)` method sense? Or just a `SpriteList.shuffle` method?

Do we need to shuffle the cards in the SpriteList at all or can we just have a separate object hold the "ordering" like a simple list (with weakrefs if that is an important factor) ? 

Sorry if these are invalid directions, but it seems like from light browsing of the SpriteList code that changing order can be "expensive" so to speak, so if all we care about is randomizing the order, changing how it is literally stored in memory seems unnecessary (Unless of course it is for reasons I am not aware of to be clear!). 

Something like this? 
```python
        # --- Create, shuffle, and deal the cards
        from random import sample
        # Sprite list with all the cards, no matter what pile they are in.
        self.card_list = arcade.SpriteList()
        # Create every card
        for card_suit in CARD_SUITS:
            for card_value in CARD_VALUES:
                card = Card(card_suit, card_value, CARD_SCALE)
                card.position = START_X, BOTTOM_Y
                self.card_list.append(card)
        self._card_order = sample(self.card_list, k=len(self.card_list))
```
Obviously more changes would be necessary to make everything use/respect that, but I think that conveys the idea. Instead of changing the order of the SpriteList in memory, change how it acts like they are ordered when we iterate over it. Maybe a subclass that has slightly different getitem/setitem logic to use in cases where you need a SpriteList you can effectively shuffle?

**Comment by mriswithe** [comment]
Just was looking at what it would take to make a subclass and I saw that there is already a swap function. So I guess that means that it did make sense, so much so that you all have already done it...... derp.

**Comment by mriswithe** [comment]
.....And `SpriteList.shuffle` already exists...... glad I could come up with something you folk hadn't already thought of /s hahaha

**Comment by mriswithe** [comment]
So far the only two code changes that seemed necessary are replace `MyGame.pull_to_top` from:
```python
    def pull_to_top(self, card):
        """ Pull card to top of rendering order (last to render, looks on-top) """
        # Find the index of the card
        index = self.card_list.index(card)
        # Loop and pull all the other cards down towards the zero end
        for i in range(index, len(self.card_list) - 1):
            self.card_list[i] = self.card_list[i + 1]
        # Put this card at the right-side/top/size of list
        self.card_list[len(self.card_list) - 1] = card
```

with this:
```python
    def pull_to_top(self, card):
        """ Pull card to top of rendering order (last to render, looks on-top) """
        # Pop from list
        self.card_list.pop(self.card_list.index(card))
        # Add to the end of the list
        self.card_list.append(card)
```

and replace the shuffle code in `MyGame.setup`
```python
        # Shuffle the cards
        for pos1 in range(len(self.card_list)):
            pos2 = random.randrange(len(self.card_list))
            self.card_list[pos1], self.card_list[pos2] = self.card_list[pos2], self.card_list[pos1]
```

with 
```python
        # Shuffle the cards
        self.card_list.shuffle()
```

I will see if I can get together a PR later on, gotta run for now

**Comment by einarf** [comment]
Fixed in 2.6.3. Be free to reopen
